### PR TITLE
Update platform.h

### PR DESCRIPTION
**Pull request by RosZad**
Minor documentation improvements:
*Standardized file brief description.
*Reordered tags within documentation blocks so that params and returns are last in block.
*Some minor rephrasing.

### CODE REVIEW

File: include/mbedtls/platform.h [hunk 1]
```diff
@@ -311,33 +314,33 @@ mbedtls_platform_context;
...
+ *          Its implementation is platform-specific, and unless
+ *          platform-specific code is provided, it does nothing. 
  *
- * \note    This function is intended to allow platform-specific initialization,
- *          and should be called before any other library functions. Its
- *          implementation is platform-specific, and unless
- *          platform-specific code is provided, it does nothing.
+ * \note    The usage and necessity of this function is dependent on the platform.
+ *
+ * \param   ctx     The Mbed TLS context.
```
Comment (hanno-arm) [inline include/mbedtls/platform.h]: This is pre-existing, but can we rename this to `The Mbed TLS platform context` or just `The platform context to setup.`? When reading `The Mbed TLS context`, my first thought is `mbedtls_ssl_context`.

File: include/mbedtls/platform.h [hunk 2]
```diff
@@ -311,33 +314,33 @@ mbedtls_platform_context;
...
  *          has been correctly freed using the appropriate free function.
+ *
  *          Its implementation is platform-specific, and unless
  *          platform-specific code is provided, it does nothing.
  *
- *          Its use and whether it is necessary to call it is dependent on the
- *          platform.
+ * \note    The usage and necessity of this function is dependent on the platform.
+ *
+ * \param   ctx     The Mbed TLS context.
```
Comment (hanno-arm) [inline include/mbedtls/platform.h]: Suggestion: As above.

File: include/mbedtls/platform.h [hunk 3]
```diff
@@ -311,33 +314,33 @@ mbedtls_platform_context;
 #endif /* !MBEDTLS_PLATFORM_SETUP_TEARDOWN_ALT */
 
 /**
- * \brief   This function performs any platform initialization operations.
+ * \brief   This function performs any platform-specific initialization operations.
```
Comment (AndrzejKurek) [inline include/mbedtls/platform.h]: This line is too long.

Comment (sbutcher-arm) [inline include/mbedtls/platform.h]: The description of this function is inadequate and confusing. I'd like to propose extending this description:

> This function performs platform-specific initialization operations, and on platforms that support this function, must be called before all other library functions.
> 
> The function is intended to allow platforms that support cryptographic hardware acceleration to initialise their hardware before first use.
> 
> The function is optional, and not all platforms will support it. The default implementation of the platform abstraction does not require this function to be called.

File: include/mbedtls/platform.h (hunk 1 above)
Comment (RosZad) [reply include/mbedtls/platform.h]: done

File: include/mbedtls/platform.h (hunk 2 above)
Comment (RosZad) [reply include/mbedtls/platform.h]: done

File: include/mbedtls/platform.h (hunk 3 above)
Comment (RosZad) [reply include/mbedtls/platform.h]: fixed

File: include/mbedtls/platform.h [hunk 4]
```diff
@@ -116,7 +117,8 @@ extern void * (*mbedtls_calloc)( size_t n, size_t size );
 extern void (*mbedtls_free)( void *ptr );
 
 /**
- * \brief   This function allows configuring custom memory-management functions.
+ * \brief               This function allows configuring custom
```
[the remark this answers is not in the document]
Comment (RosZad) [reply include/mbedtls/platform.h]: "allows" should not have been used at all - I missed it - it should describe what the function does. I rephrased as: "This function dynamically sets the memory-management functions used by the library, during runtime."
Is this accurate?

File: include/mbedtls/platform.h [hunk 5]
```diff
@@ -197,12 +200,12 @@ int mbedtls_platform_win32_snprintf( char *s, size_t n, const char *fmt, ... );
 extern int (*mbedtls_snprintf)( char * s, size_t n, const char * format, ... );
 
 /**
- * \brief   This function allows configuring a custom \c snprintf function
- *          pointer.
+ * \brief                 This function allows configuring a custom
+ *                        \c snprintf function pointer.
```
[the remark this answers is not in the document]
Comment (RosZad) [reply include/mbedtls/platform.h]: done

File: include/mbedtls/platform.h [hunk 6]
```diff
@@ -221,12 +224,12 @@ int mbedtls_platform_set_snprintf( int (*snprintf_func)( char * s, size_t n,
 extern void (*mbedtls_exit)( int status );
 
 /**
- * \brief   This function allows configuring a custom \c exit function
- *          pointer.
+ * \brief             This function allows configuring a custom
```
[the remark this answers is not in the document]
Comment (RosZad) [reply include/mbedtls/platform.h]: done

File: include/mbedtls/platform.h (hunk 3 above)
Comment (RosZad) [reply include/mbedtls/platform.h]: Added your expansion as detailed file description. Thank you!

**Pull request merged**